src/de/variant_de.rs

```diff
@@ -0,0 +1,150 @@
+use crate::safearray::get_string_array;
+use failure::{bail, Error};
+use log::debug;
+use serde::{de, forward_to_deserialize_any, Deserialize};
+use std::fmt;
+use crate::variant::Variant;
+
+struct SeqAccess {
+    data: Vec<Variant>,
+    i: usize,
+}
+
+impl<'de> de::SeqAccess<'de> for SeqAccess {
+    type Error = crate::error::Error;
+
+    fn next_element_seed<T>(&mut self, seed: T) -> Result<Option<T::Value>, Self::Error>
+        where
+            T: de::DeserializeSeed<'de>,
+    {
+        if self.i >= self.data.len() {
+            return Ok(None);
+        }
+
+        let res: Variant = self.data.swap_remove(self.i);
```

Comment:
@omerbenamram BTW this is a terrible bug (since `swap_remove`, as the name suggest, does not preserve the order, so we basically just return some random half-list).

It works because the array used in the tests is of length 1 (🙄 ).

I fixed it in a newer branch.

Reply:
מצחיק